lib/util/platform.js

```diff
@@ -550,6 +551,146 @@ shaka.util.Platform = class {
+          Windows.Media.Protection.ProtectionCapabilityResult;
+      // isTypeSupported may return "maybe", which means the operation is not
+      // completed. This means we need to retry
+      // https://learn.microsoft.com/en-us/uwp/api/windows.media.protection.protectioncapabilityresult?view=winrt-22621
+      let result = null;
+      try {
+        const type =
+            'video/mp4;codecs="hvc1,mp4a";features="decode-res-x=3840,' +
+            'decode-res-y=2160,decode-bitrate=20000,decode-fps=30,' +
+            'decode-bpc=10,display-res-x=3840,display-res-y=2160,' +
+            'display-bpc=8"';
+        const keySystem = 'com.microsoft.playready.recommendation';
+        do {
+          result = protectionCapabilities.isTypeSupported(type, keySystem);
+        } while (result === protectionResult.maybe);
+      } catch (e) {
+        result = protectionResult.notSupported;
+      }
+      if (result === protectionResult.probably) {
+        maxResolution.width = 3840;
+        maxResolution.height = 2160;
+      } else {
+        maxResolution.width = 1920;
+        maxResolution.height = 1080;
+      }
+    } else if (Platform.isWebOS()) {
+      try {
+        const deviceInfo =
+        /** @type {{screenWidth: number, screenHeight: number}} */(
+            JSON.parse(window.PalmSystem.deviceInfo));
+        // WebOS has always been able to do 1080p.  Assume a 1080p limit.
+        maxResolution.width = Math.max(1920, deviceInfo.screenWidth);
+        maxResolution.height = Math.max(1080, deviceInfo.screenHeight);
+      } catch (e) {
+        shaka.log.alwaysWarn('WebOS: Error detecting screen size, default ' +
+            'screen size 1920x1080.');
+        maxResolution.width = 1920;
+        maxResolution.height = 1080;
+      }
+    } else if (Platform.isHisense()) {
+      // eslint-disable-next-line new-cap
+      if (window.Hisense_Get4KSupportState &&
+          // eslint-disable-next-line new-cap
+          window.Hisense_Get4KSupportState()) {
+        maxResolution.width = 3840;
+        maxResolution.height = 2160;
+      } else {
+        maxResolution.width = 1920;
+        maxResolution.height = 1080;
+      }
+    } else if (Platform.isPS4()) {
+      let supports4K = false;
+      try {
+        const result = await window.msdk.device.getDisplayInfo();
+        supports4K = result.resolution === '4K';
+      } catch (e) {
+        shaka.log.alwaysWarn('PS4: Failed to get the display info:', e);
+      }
+      if (supports4K) {
+        maxResolution.width = 3840;
+        maxResolution.height = 2160;
+      } else {
+        maxResolution.width = 1920;
+        maxResolution.height = 1080;
+      }
+    } else if (Platform.isPS5()) {
+      let supports4K = false;
+      try {
+        const result = await window.msdk.device.getDisplayInfoImmediate();
+        supports4K = result.resolution === '4K';
+      } catch (e) {
+        shaka.log.alwaysWarn('PS5: Failed to get the display info:', e);
+      }
```

Comment:
What's the difference between `getDisplayInfo()` & `getDisplayInfoImmediate()`?

Reply:
Actually none, but the API name is differently on PS4 and PS5.